File: 110_Balanced_Binary_Tree/110_1.c
#define abs(x, y) ((x) - (y) >= 0 ? (x) - (y) : (y) - (x))
#define max(x, y) ((x) > (y) ? (x) : (y))

int depth(struct TreeNode *root) {
    if (!root) return 0;
    int left = depth(root->left);
    int right = depth(root->right);
    if (left == -1 || right == -1 || abs(left, right) > 1) return -1;
    return 1 + max(left, right);
}

bool isBalanced(struct TreeNode* root){
    return depth(root) != -1;
}

File: 110_Balanced_Binary_Tree/110_1.java
class Solution {
    public boolean isBalanced(TreeNode root) {
        if (root == null) return true;
        int leftDepth = maxDepth(root.left);
        int rightDepth = maxDepth(root.right);
        boolean rootBalanced = Math.abs(leftDepth - rightDepth) <= 1;
        return rootBalanced && isBalanced(root.left) && isBalanced(root.right);
    }

    private int maxDepth(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
    }
}